loadtest/control/actions.go

```diff
@@ -0,0 +1,420 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package control
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	"github.com/mattermost/mattermost-load-test-ng/loadtest/user"
+	"github.com/mattermost/mattermost-server/v5/model"
+)
+
+// UserActionResponse is a structure containing information about the result
+// of running a UserAction.
+type UserActionResponse struct {
+	// Info contains a string with information about the action
+	// execution.
+	Info string
+	// Err contains an error when the action failed.
+	Err error
+}
+
+// UserAction is a function that simulates a specific behaviour for the provided
+// user.User. It returns a UserActionResponse.
+type UserAction func(user.User) UserActionResponse
+
+func SignUp(u user.User) UserActionResponse {
```

Comment:
These became public functions now and need godocs. We can do it later if you prefer.

Reply:
Yep, it's going to be a separate ticket given it's a non trivial amount of docs.